### ISSUE TITLE
solve!() breaks with BoundsError when decision variable multiplied by 0

### ISSUE THREAD

**Issue by MFairley** [jump-dev/MosekTools.jl#56]
Here's a simple MWE that causes the solver to break.

```julia
using Convex
using Mosek
using MosekTools

x = Variable()
y = Variable()
problem = minimize(0.0 * x + y)
solve!(problem, () -> Mosek.Optimizer())
```

```
ERROR: LoadError: BoundsError: attempt to access 2-element Array{Float64,1} at index [3]
Stacktrace:
 [1] getindex at ./array.jl:809 [inlined]
 [2] getindex at ./multidimensional.jl:557 [inlined]
 [3] variable_primal at /Users/mfairley/.julia/packages/MosekTools/prFdq/src/attributes.jl:44 [inlined]
 [4] variable_primal at /Users/mfairley/.julia/packages/MosekTools/prFdq/src/attributes.jl:54 [inlined]
 [5] get(::MosekModel, ::MathOptInterface.VariablePrimal, ::MathOptInterface.VariableIndex) at /Users/mfairley/.julia/packages/MosekTools/prFdq/src/attributes.jl:261
 [6] get!(::Array{Float64,1}, ::MosekModel, ::MathOptInterface.VariablePrimal, ::Array{MathOptInterface.VariableIndex,1}) at /Users/mfairley/.julia/packages/MosekTools/prFdq/src/attributes.jl:268
 [7] get at /Users/mfairley/.julia/packages/MosekTools/prFdq/src/attributes.jl:275 [inlined]
 [8] get(::MathOptInterface.Utilities.CachingOptimizer{MosekModel,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}, ::MathOptInterface.VariablePrimal, ::Array{MathOptInterface.VariableIndex,1}) at /Users/mfairley/.julia/packages/MathOptInterface/k7UUH/src/Utilities/cachingoptimizer.jl:621
 [9] get(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{MosekModel,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}}, ::MathOptInterface.VariablePrimal, ::Array{MathOptInterface.VariableIndex,1}) at /Users/mfairley/.julia/packages/MathOptInterface/k7UUH/src/Bridges/bridge_optimizer.jl:818
 [10] moi_populate_solution!(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{MosekModel,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}}, ::Problem{Float64}, ::OrderedCollections.OrderedDict{UInt64,Convex.AbstractVariable}, ::Dict{Convex.ConicConstr,Constraint}, ::Array{Convex.ConicConstr,1}, ::Dict{UInt64,Array{MathOptInterface.VariableIndex,1}}, ::Array{MathOptInterface.ConstraintIndex{MathOptInterface.VectorAffineFunction{Float64},MathOptInterface.Zeros},1}) at /Users/mfairley/.julia/packages/Convex/aYxJA/src/solution.jl:288
 [11] solve!(::Problem{Float64}, ::MosekModel; check_vexity::Bool, verbose::Bool, warmstart::Bool, silent_solver::Bool) at /Users/mfairley/.julia/packages/Convex/aYxJA/src/solution.jl:249
 [12] solve! at /Users/mfairley/.julia/packages/Convex/aYxJA/src/solution.jl:223 [inlined]
 [13] solve!(::Problem{Float64}, ::var"#1#2"; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /Users/mfairley/.julia/packages/Convex/aYxJA/src/solution.jl:192
 [14] solve!(::Problem{Float64}, ::Function) at /Users/mfairley/.julia/packages/Convex/aYxJA/src/solution.jl:191
 [15] top-level scope at /Users/mfairley/Projects/AdaptiveSurveillance/test/cvx_mwe_bug.jl:8
 [16] include(::Function, ::Module, ::String) at ./Base.jl:380
 [17] include(::Module, ::String) at ./Base.jl:368
 [18] exec_options(::Base.JLOptions) at ./client.jl:296
 [19] _start() at ./client.jl:506
in expression starting at /Users/mfairley/Projects/AdaptiveSurveillance/test/cvx_mwe_bug.jl:8
```

**Comment by blegat** [comment]
I guess it's due to https://github.com/jump-dev/MosekTools.jl/issues/43

**Comment by blegat** [comment]
It should be fixed in the new release v0.9.4 of this package

**Comment by MFairley** [comment]
Yes, it's working now. Thanks!
